### DISCUSSION TITLE
webpack injects `document` into bundle, which fails eval

### DISCUSSION THREAD

**Discussion by rygine** [MetaMask/snaps#2135] [Q&A] [answered]
working build on `main` branch here: https://github.com/xmtp/snap

failing build on branch `rygine/upgrade-sdk`. the only change is upgrading our SDK version.

i'm seeing the following error during the eval stage: `ReferenceError: document is not defined`

after tracking down where `document` is used, i discovered that it appears to be injected into the bundle via the `JsonpChunkLoadingRuntimeModule` class used by `webpack` ([source](https://github.com/webpack/webpack/blob/4a2662381ed8e6a9d849b494511459f5858d4e82/lib/web/JsonpChunkLoadingRuntimeModule.js#L68)).

nothing has changed in our SDK with regards to the use of `document` or `window.document`.

please advise. thanks!

**Comment by Mrtenz** [comment, marked as answer]
Does this solve the problem? This should disable `JsonpChunkLoadingRuntimeModule`.

```ts
import type { SnapConfig } from '@metamask/snaps-cli';
import { merge } from '@metamask/snaps-cli';

const config: SnapConfig = {
  // ...
  customizeWebpackConfig: (baseConfig) => {
    return merge(baseConfig, {
      output: {
        chunkFormat: 'commonjs',
      },
    });
  },
};

export default config;
```

**Reply by rygine** [reply]
that fixed it, thanks!

**Reply by Mrtenz** [reply]
Glad to hear! I'll submit a PR to get this enabled by default.
